File: test/crawl-workflow.test.py
from pathlib import Path
import unittest


ROOT = Path(__file__).resolve().parents[1]


class CrawlWorkflowTest(unittest.TestCase):
    def test_goofish_uses_china_only_keywords(self):
        workflow = (ROOT / ".github" / "workflows" / "crawl.yml").read_text(encoding="utf-8")

        self.assertIn("default: joongna,bunjang,guheyo,mercari,goofish", workflow)
        self.assertIn(
            "CHINA_KEYWORDS: ${{ inputs.china_keywords || '三亩S80,三亩s58m,MetaKeebs,realforce' }}",
            workflow,
        )
        self.assertIn('--keywords="$CHINA_KEYWORDS" --markets="goofish"', workflow)
        self.assertIn('--keywords="$KEYWORDS" --markets="$non_china_markets"', workflow)
        self.assertIn('if ! node crawl.js --keywords="$CHINA_KEYWORDS" --markets="goofish"', workflow)
        self.assertIn('::warning::Goofish crawl failed; keeping the previous data and manifest error.', workflow)

    def test_crawl_panel_selects_goofish(self):
        panel = (ROOT / "index.html").read_text(encoding="utf-8")

        self.assertIn(
            '<input name="crawl-market" type="checkbox" value="goofish" checked /> 🇨🇳 Goofish',
            panel,
        )


if __name__ == "__main__":
    unittest.main()

File: test/crawl-workflow.test.js
import test from "node:test";
import assert from "node:assert/strict";
import { readFile } from "node:fs/promises";

const workflowPath = new URL("../.github/workflows/crawl.yml", import.meta.url);
const panelPath = new URL("../index.html", import.meta.url);

test("workflow crawls Yahoo Auctions normally and isolates Goofish failures", async () => {
  const workflow = await readFile(workflowPath, "utf8");

  assert.match(workflow, /default: joongna,bunjang,guheyo,mercari,yahoo-auctions,goofish/);
  assert.match(workflow, /--keywords="\$KEYWORDS" --markets="\$non_china_markets"/);
  assert.match(workflow, /if ! node crawl\.js --keywords="\$KEYWORDS" --markets="goofish"/);
  assert.match(
    workflow,
    /::warning::Goofish crawl failed; keeping the previous data and manifest error\./,
  );
});

test("crawl panel selects Yahoo Auctions and Goofish by default", async () => {
  const panel = await readFile(panelPath, "utf8");

  assert.match(
    panel,
    /<input name="crawl-market" type="checkbox" value="goofish" checked \/>\s*🇨🇳\s*Goofish/,
  );
  assert.match(
    panel,
    /<input name="crawl-market" type="checkbox" value="yahoo-auctions" checked \/>\s*🇯🇵\s*Yahoo! Auctions/,
  );
});
